### PR TITLE
[SPARK-26192][MESOS] Retrieve enableFetcherCache option from submission for driver URIs

### PR DESCRIPTION
**Pull request by mwlon**
## What changes were proposed in this pull request?

Retrieve enableFetcherCache option from submission conf rather than dispatcher conf. This resolves some confusing behavior where Spark drivers currently get this conf from the dispatcher, whereas Spark executors get this conf from the submission. After this change, the conf will only need to be specified once.

## How was this patch tested?

With (updated) existing tests.

### CODE REVIEW

File: resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala [hunk 1]
```diff
@@ -104,7 +104,7 @@ private[mesos] class MesosSubmitRequestServlet(
     val driverMemory = sparkProperties.get(config.DRIVER_MEMORY.key)
     val driverMemoryOverhead = sparkProperties.get(config.DRIVER_MEMORY_OVERHEAD.key)
     val driverCores = sparkProperties.get(config.DRIVER_CORES.key)
-    val name = request.sparkProperties.getOrElse("spark.app.name", mainClass)
+    val name = sparkProperties.getOrElse("spark.app.name", mainClass)
```
Comment (dongjoon-hyun) [inline resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala]: Is this related to this PR?

Comment (mwlon) [reply resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala]: No - should I reset it?

Comment (dongjoon-hyun) [reply resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala]: Yep, please.

Comment (mwlon) [reply resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala]: undid it

Comment (dongjoon-hyun) [reply resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala]: Thanks!

File: resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala [hunk 2]
```diff
@@ -427,6 +426,7 @@ private[spark] class MesosClusterScheduler(
   }
 
   private def getDriverUris(desc: MesosDriverDescription): List[CommandInfo.URI] = {
+    val useFetchCache = desc.conf.get(config.ENABLE_FETCHER_CACHE)
```
Comment (dongjoon-hyun) [inline resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala]: Can we make this in a backward-compatible way? For example, if the configuration is not given from the submission, Spark had better work in the old way.

**Comment by dongjoon-hyun** [comment]
ok to test

File: resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala (hunk 2 above)
Comment (mwlon) [reply resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala]: Ok, set it to turn fetcher cache on if specified in either dispatcher or submission

File: resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala [hunk 3]
```diff
@@ -427,6 +426,7 @@ private[spark] class MesosClusterScheduler(
   }
 
   private def getDriverUris(desc: MesosDriverDescription): List[CommandInfo.URI] = {
+    val useFetchCache = desc.conf.get(config.ENABLE_FETCHER_CACHE) || conf.get(config.ENABLE_FETCHER_CACHE)
```
Comment (dongjoon-hyun) [inline resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala]: Could you fix the Scaly style error at this line? You can check with the following command in your PR directory.
```
$ dev/scalastyle
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
